### PR TITLE
Moved methods to packing utils

### LINKED ISSUE #1457: Move common packing methods into packing utils
Some methods are used by different packing algos and can be moved into packing utils

### PR DESCRIPTION
**Pull request by avflor**
fixes #1457

### CODE REVIEW

**Comment by avflor** [comment]
@billonahill Can you please take a look at this? It is some refactoring for the RR repack

File: heron/packing/src/java/com/twitter/heron/packing/PackingUtils.java [hunk 1]
```diff
@@ -173,4 +174,47 @@ public static double increaseBy(double value, int paddingPercentage) {
...
+
+  /**
+   * Identifies which components need to be scaled down
+   *
+   * @return Map &lt; component name, scale down factor &gt;
+   */
+  public static Map<String, Integer> getComponentsToScaleDown(Map<String,
+      Integer> componentChanges) {
+    Map<String, Integer> componentsToScaleDown = new HashMap<String, Integer>();
+    for (Map.Entry<String, Integer> entry : componentChanges.entrySet()) {
```
Comment (billonahill) [inline heron/packing/src/java/com/twitter/heron/packing/PackingUtils.java]: can we iterate over the keySet() here instead? The naming helps it tend to be more descriptive than using entry.getKey() and entry.getValue().

File: heron/packing/src/java/com/twitter/heron/packing/PackingUtils.java [hunk 2]
```diff
@@ -173,4 +174,47 @@ public static double increaseBy(double value, int paddingPercentage) {
...
+   */
+  public static int allocateNewContainer(ArrayList<Container> containers, Resource capacity,
+                                         int paddingPercentage) {
+    containers.add(new Container(capacity, paddingPercentage));
+    return containers.size();
+  }
+
+  /**
+   * Identifies which components need to be scaled down
+   *
```
Comment (billonahill) [inline heron/packing/src/java/com/twitter/heron/packing/PackingUtils.java]: pls specify what the String and Integer in the map represent.

**Comment by billonahill** [comment]
Looks good, just one nit.

**Comment by billonahill** [comment]
👍 once ci passes

**Pull request merged**